File: WebClient SDK/src/default/documents/hideEmptyIncomingReferencesControl.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';
import { DefaultFormMainViewModel } from 'tessa/ui/cards/forms';
import { AutoCompleteTableViewModel } from 'tessa/ui/cards/controls';
import { Visibility } from 'tessa/platform';

export class HideEmptyIncomingReferencesControl extends CardUIExtension {
  public initialized(context: ICardUIExtensionContext) {
    const model = context.model;
    const view = model.mainForm as DefaultFormMainViewModel;

    if (!view || !(view instanceof DefaultFormMainViewModel)) {
      return;
    }

    const block = view.blocks.find(x => x.name === 'RefsBlock');
    if (!block) {
      return;
    }

    const control = block.controls.find(x => x.name === 'IncomingRefsControl');
    const autoComplete = control as AutoCompleteTableViewModel;

    if (
      autoComplete &&
      autoComplete instanceof AutoCompleteTableViewModel &&
      autoComplete.items.length === 0
    ) {
      autoComplete.controlVisibility = Visibility.Collapsed;
    }
  }
}

File: Extensions/Tessa.Extensions.Default.Client/Documents/HideEmptyIncomingReferencesControl.cs
using System.Linq;
using System.Threading.Tasks;
using Tessa.UI.Cards;
using Tessa.UI.Cards.Controls.AutoComplete;
using Tessa.UI.Cards.Forms;

namespace Tessa.Extensions.Default.Client.Documents
{
    public sealed class HideEmptyIncomingReferencesControl : CardUIExtension
    {
        public override async Task Initialized(ICardUIExtensionContext context)
        {
            ICardModel model = context.Model;

            if (!(model.MainForm is DefaultFormMainViewModel view))
            {
                return;
            }

            IBlockViewModel block = view.Blocks.FirstOrDefault(x => x.Name == "RefsBlock");
            if (block == null)
            {
                return;
            }

            IControlViewModel control = block.Controls.FirstOrDefault(x => x.Name == "IncomingRefsControl");

            if (control is AutoCompleteTableViewModel autoComplete && autoComplete.Items.Count == 0)
            {
                control.ControlVisibility = System.Windows.Visibility.Collapsed;
            }
        }
    }
}
